Save a Copy: AssertionError: too many color changes, (11986) out of bounds (0, 255)

File - Save a Copy
Select ZIP
Check all file format 
wait for 5 minutes 

Ink/Stitch Version: v3.0.1 (windows)


Traceback (most recent call last):
  File "inkstitch.py", line 84, in <module>
  File "inkex\base.py", line 250, in run
  File "lib\extensions\zip.py", line 68, in effect
  File "lib\output.py", line 109, in write_embroidery_file
  File "pyembroidery\EmbPattern.py", line 1727, in static_write
  File "pyembroidery\EmbPattern.py", line 1644, in write_embroidery
  File "pyembroidery\PecWriter.py", line 32, in write
  File "pyembroidery\PecWriter.py", line 40, in write_pec
  File "pyembroidery\PecWriter.py", line 64, in write_pec_header
AssertionError: too many color changes, (11986) out of bounds (0, 255)

Maybe the _Desity Map_ function is the problem?
![satoshi-szobor-simple-v1](https://github.com/inkstitch/inkstitch/assets/63737882/122c2207-e8f7-487b-9946-cd326f9298a0)

--- a/lib/extensions/density_map.py
+++ b/lib/extensions/density_map.py
@@ -3,11 +3,11 @@
 # Copyright (c) 2010 Authors
 # Licensed under the GNU GPL version 3.0 or later.  See the file LICENSE for details.
 
+import inkex
 import numpy as np
 from scipy.spatial import KDTree
 
-import inkex
-
+from ..commands import add_layer_commands
 from ..i18n import _
 from ..stitch_plan import stitch_groups_to_stitch_plan
 from ..svg import PIXELS_PER_MM
@@ -70,6 +70,7 @@ def reset_density_plan(svg):
             INKSCAPE_GROUPMODE: 'layer'
         })
         svg.append(layer)
+        add_layer_commands(layer, ["ignore_layer"])
     else:
         # delete old density plan
         del layer[:]

